pkg/cortex/modules.go

```diff
@@ -165,22 +165,27 @@ func (t *Cortex) initRuntimeConfig() (services.Service, error) {
 }
 
 func (t *Cortex) initOverrides() (serv services.Service, err error) {
-	t.Overrides, err = validation.NewOverrides(t.Cfg.LimitsConfig, tenantLimitsFromRuntimeConfig(t.RuntimeConfig))
+	// If runtime configuration isn't enabled it doesn't make sense to configure per-tenant
+	// limits. Any consumers of TenantLimits will handle it being `nil` correctly.
+	if t.RuntimeConfig != nil {
```

Comment:
If you want to be able to override `t.TenantLimits` in project based on Cortex, then we need to set `t.TenantLimits` in `initRuntimeConfig` (which also makes sense, since we only need to set it when actually using runtime config). `initOverrides` will only use, but not set, `t.TenantLimits`.

Then in your project, you can add dependency from `Overrides` to `CustomTenantLimits`, with its own init method, which will be called *before* `initOverrides`, so it has a chance to modify `t.TenantLimits`.